go/mysql/binlog_event_compression.go

```diff
@@ -89,23 +89,14 @@ var (
 	// allocations and GC overhead so this pool allows us to handle
 	// concurrent cases better while still scaling to 0 when there's no
 	// usage.
-	statefulDecoderPool sync.Pool
+	statefulDecoderPool = &decoderPool{}
 )
 
 func init() {
 	var err error
 	statelessDecoder, err = zstd.NewReader(nil, zstd.WithDecoderConcurrency(0))
 	if err != nil { // Should only happen e.g. due to ENOMEM
-		log.Errorf("Error creating stateless decoder: %v", err)
-	}
-	statefulDecoderPool = sync.Pool{
-		New: func() any {
-			d, err := zstd.NewReader(nil, zstd.WithDecoderMaxMemory(zstdInMemoryDecompressorMaxSize))
-			if err != nil { // Should only happen e.g. due to ENOMEM
-				log.Errorf("Error creating stateful decoder: %v", err)
-			}
-			return d
-		},
+		panic(fmt.Errorf("failed to create stateless decoder: %v", err))
```

Comment:
Should we panic here and prevent the process from launching? Option is for the `decompress()` to return an  error if `statefulDecoderPool` failed to initialize.

Reply:
It's better to panic, I think, as this is only allocated once at init() time. If we can't initialize it, we won't be able to run properly. The way the code was previously, it would just continually log errors w/o being able to do much. This should only ever fail if you're very low on memory (ENOMEM).

Reply:
Only for compressed payloads, right? Is it documented that this failure can only happen for `ENOMEM`. In case there is some other issue with `zstd` preventing the executable from launching seems excessive and risky.

Reply:
Yeah, that's true.... maybe I'll just go back to logging like I was before then. It also has the benefit of limiting the changes.

Reply:
I went ahead and got rid of the panic and init(): https://github.com/vitessio/vitess/pull/17042/commits/e7191f0be5a6a26420ef39128ba3e2f47d997833

No reason for us NOT to try initializing it again the next time we retry...